crates/net/discv4/src/lib.rs

```diff
@@ -464,9 +508,14 @@ pub struct Discv4Service {
     received_pongs: PongTable,
     /// Interval used to expire additionally tracked nodes
     expire_interval: Interval,
+    /// Mirrors discv5 kbuckets. Set if node is run with discv5, to support downgrading to discv4.
```

Comment:
~~what does downgrading to discv4 mean?~~

ah I think I understand this now.

But I'd need a few more docs to completely understand it.
just one though I have, perhaps it's possible to move this to a new type that wraps `discv5: M` and the service?